### ISSUE TITLE
Custom MFA handler doesn't seem to work?

### ISSUE THREAD

**Issue by cyberjunky** [matin/garth#86]
Opening a new ticket since the old one was closed and full of api changes.

It seems the garth.login method always returns None when a MFA enabled account is used and return_on_mfa=True

```
  result = self.garth.login(
      self.username, self.password, return_on_mfa=True
  )
```
I believe garth.login -> client.login which is http.login which points to sso.login
But http.login doesn't return the same args

Can you shed a light, see what I do wrong, or if there is a bug?

variable title is holding "MFA" in sso.login, but the returned dict never ends up in garth.login (= None)

Running with garth 0.5.2 btw, trying to adapt my example.py

**Comment by matin** [comment]
I'll take a look

**Comment by cyberjunky** [comment]
Any luck in finding if there is an issue, and/or fix?

**Comment by cyberjunky** [comment]
@matin I got it working, see PR https://github.com/matin/garth/pull/92 hope you can merge it, if it can be optimized please do ofcourse!

**Comment by nouser2013** [comment]
Out of curiosity, is there any way to use this in a web app context? Assuming I want to use GARTH to get an OAuth token set from Garmin, but I have the 2 step process where the GARTH http client and its ```last_resp``` content get destroyed after the ```sso.login()``` returns.

In other words, I'd like to have one web form for the users email and password, and then, after some time, I'd like to display a second web form expecting the MFA token input.

I tried ```pickle```ing the http client for fun and move it across the HTML form (yes, very bad idea, I know), but it also didn't work, of course.

Thanks.

**Comment by matin** [comment]
closed by merging #92

**Comment by matin** [comment]
@nouser2013 this change should enable you to do that

**Comment by cyberjunky** [comment]
Yes this works, there is one thing though, after resume and supplying the MFA code, I need to call the login() call again, nothing big, and cannot harm, could not find the cause/how to fix in Garth.
